programs/drift/src/math/liquidation.rs

```diff
@@ -486,3 +493,22 @@ pub fn get_liquidation_order_params(
 
     Ok(order_params)
 }
+
+pub fn get_liquidation_fee(
+    base_liquidation_fee: u32,
+    max_liquidation_fee: u32,
+    last_active_user_slot: u64,
+    current_slot: u64,
+) -> DriftResult<u32> {
+    let slots_elapsed = current_slot.safe_sub(last_active_user_slot)?;
+    if slots_elapsed < LIQUIDATION_FEE_ADJUST_GRACE_PERIOD_SLOTS {
+        return Ok(base_liquidation_fee);
+    }
+
+    let liquidation_fee = base_liquidation_fee.safe_add(
+        slots_elapsed
+            .safe_mul(LIQUIDATION_FEE_INCREASE_PER_SLOT.cast::<u64>()?)?
+            .cast::<u32>()?,
```

Comment:
can this ever overlow? might just do unwrap_or(u32::MAX)

Reply:
yeah, good call. done.